plugins/node/opentelemetry-instrumentation-mysql/src/instrumentation.ts

```diff
@@ -142,11 +166,39 @@ export class MySQLInstrumentation extends InstrumentationBase<
           'getConnection',
           thisPlugin._patchGetConnection(pool, format)
         );
+        thisPlugin._wrap(pool, 'end', thisPlugin._patchPoolEnd(pool));
+        thisPlugin._setPoolcallbacks(pool, thisPlugin, '');
 
         return pool;
       };
     };
   }
+  private _patchPoolEnd(pool: any) {
+    return (originalPoolEnd: Function) => {
+      const thisPlugin = this;
+      diag.debug('MySQLInstrumentation#patch: patched mysql pool end');
+      return function end(callback?: unknown) {
+        const nAll = (pool as any)._allConnections.length;
+        const nFree = (pool as any)._freeConnections.length;
+        const nUsed = nAll - nFree;
+        const poolName = getPoolName(pool);
+
+        for (let i = 0; i < nUsed; i++) {
+          thisPlugin._connectionsUsage.add(-1, {
+            state: 'used',
+            name: poolName,
+          });
+        }
+        for (let i = 0; i < nFree; i++) {
+          thisPlugin._connectionsUsage.add(-1, {
+            state: 'idle',
+            name: poolName,
+          });
+        }
```

Comment:
you can straight add and reduce the amount of the connections without these for loops.

Reply:
Yeah I didn't catch this before but `thisPlugin._connectionsUsage.add(-nUsed, { state: 'used', name: poolName })` should have the same effect as the top loop. The bottom loop can be similarly replaced.

Reply:
You're both right. my mistake. I got too used to the addition of 1/-1 .....

Reply:
Fixed in [3b78a86](https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1220/commits/3b78a867c9ba471a069813c21993448cac723786) :slightly_smiling_face: